### ISSUE TITLE
Incorrect encoding of the <location> tag in the XSPF playlists

### ISSUE THREAD

**Issue by iamalexei** [Taiko2k/Tauon#1088]
Tauon Music Box should export XSPF playlists with the `<location>` tag encoded according to the Spec ([6.2 Relative paths
](https://www.xspf.org/spec#62-relative-paths)):

> Generators should take extra care to ensure that relative paths are correctly encoded. Do:
> 
> `<location>My%20Song.flac</location>`
> 
> Don’t:
> 
> `<location>My Song.flac</location>`

**Comment by prathamnagpure** [comment]
```python 
xport.write('      <location>' + escape(path) + '</location>\n')
```
and escape is 
```python
def escape(data, entities={}):
    """Escape &, <, and > in a string of data.

    You can escape other strings of data by passing a dictionary as
    the optional entities parameter.  The keys and values must all be
    strings; each key will be replaced with its corresponding value.
    """

    # must do ampersand first
    data = data.replace("&", "&amp;")
    data = data.replace(">", "&gt;")
    data = data.replace("<", "&lt;")
    if entities:
        data = __dict_replace(data, entities)
    return data
```

should [urllib.parse.quote](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) `urllib.parse.quote(path)`be used instead of `escape(path)`?

**Comment by Taiko2k** [comment]
> should [urllib.parse.quote](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) `urllib.parse.quote(path)`be used instead of `escape(path)`?

Yeah seems so.

**Comment by prathamnagpure** [comment]
I wanted to fix this issue but I did not understand how to build and test and currently I am on windows.

**Comment by Taiko2k** [comment]
There's a guide in the github wiki, but it's a little involved.

**Comment by iamalexei** [comment]
> I wanted to fix this issue but I did not understand how to build and test and currently I am on windows.

I don't know if it's possible on Windows, but on Linux I was able to modify _t_main.py_ in the already installed Tauon to test your solution. It seems that it works: Tauon creates XSPF playlists with the properly encoded `<location>` tag.

**Comment by Taiko2k** [comment]
The source isn't easily edited in the Windows build as it's bundled into an exe.

**Comment by C0rn3j** [comment]
@iamalexei Are you able to send a PR since you were able to fix and test it?

**Comment by iamalexei** [comment]
@C0rn3j I thought it'd be polite to wait for a PR from @prathamnagpure since I'm not the one who came up with the solution.

**Comment by C0rn3j** [comment]
> @C0rn3j I thought it'd be polite to wait for a PR from @prathamnagpure since I'm not the one who came up with the solution.

Ah it was originally their fix :)

Well, enough time has passed and no PR was sent since, you can always add them as a coauthor on the commit I believe.

**Comment by iamalexei** [comment]
@C0rn3j just checked, the issue has already been fixed in 0a59adc04af6813d1bae9d737754af8d2ecfd98c
